templates/_db-migrate.tpl

```diff
@@ -0,0 +1,75 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Spec template for DB migration pre- and post-install/upgrade jobs.
+*/}}
+{{- define "mastodon.dbMigrateJob" -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  {{- if .preDeploy }}
+  name: {{ include "mastodon.fullname" . }}-db-pre-migrate
+  {{- else }}
+  name: {{ include "mastodon.fullname" . }}-db-post-migrate
+  {{- end }}
+  labels:
+    {{- include "mastodon.labels" . | nindent 4 }}
+  annotations:
+    {{- if .preDeploy }}
+    "helm.sh/hook": pre-upgrade
+    {{- else }}
+    "helm.sh/hook": post-install,post-upgrade
```

Comment:
Does this need to be `post-install`? If we do it this way, then on the first install of the chart the sidekiq/web pods will start, then the migration will run, but there will be a time where the schema is empty. `pre-install` should be better here?

Also on initial install, we do not recommend to run the migrations (it will run every DDL one by one), but to use `db:setup` which will load it all at once. So doing that is probably better?

Reply:
Ah, it's been a while since I did this from scratch, and forgot about `db:setup`. So that should be run before anything is deployed for the schema, then the pods can be created. I will make that change.

Then I have two questions:
- Is `db:setup` idempotent? If it is, we can just blindly run it every time and not worry about it. Otherwise, we will have to create some kind of logic to only have it run if we can detect that the database schema is already present.
- If `db:setup` is run first on an initial install, is there a problem with running `db:migrate` as a `post-install` hook? I figure it will simply do nothing.

Reply:
> Is db:setup idempotent? 

It should, but I just wiped by dev database running it again, so it looks like it does not? Need more tests on this

> If db:setup is run first on an initial install, is there a problem with running db:migrate as a post-install hook? I figure it will simply do nothing.

Yes, you can run the migrations and nothing will happen

Reply:
I don't think you can run migration pre-install. Because you won't have a database at that point 🤔

Reply:
This is true definitely. That's why I've since updated this PR to include an additional job that will run `db:prepare` as a `pre-install` hook to set up the the database schema (if it's not already there) before anything is installed.

Reply:
@renchap Ok, I think this is now in an OK state.

It runs `db:prepare` on an install, but only when the user is not using the included postgres (since there's no way to install a dependency before a pre-install hook). Added comments in a couple places in the values file to point this out.